lcls_tools/common/devices/wire.py

```diff
@@ -0,0 +1,281 @@
+
+    @property
+    def x_wire_outer(self):
+        return self.controls_information.PVs.xwireouter.get()
+    @x_wire_outer.setter
+    def x_wire_outer(self, val: int) -> None:
+        if not isinstance(val, int):
+            print("Value must be an int")
+            return
+        self.controls_information.PVs.xwireouter.put(value = val)
+
+    @property
+    def use_y_wire(self):
+        return self.metadata.PVs.useywire.get()
+    @use_y_wire.setter
+    def use_y_wire(self, val: int) -> None:
+        if not (val == 1 or val == 0):
+            print("Value must be 0 or 1")
+            return
+        self.controls_information.PVs.useywire.put(value = val)
+
+    @property
+    def y_wire_inner(self):
+        return self.controls_information.PVs.ywireinner.get()
+    @y_wire_inner.setter
+    def y_wire_inner(self, val: int) -> None:
+        if not isinstance(val, int):
+            print("Value must be an int")
+            return
+        self.controls_information.PVs.ywireinner.put(value = val)
+
+    @property
+    def y_wire_outer(self):
+        return self.controls_information.PVs.ywireouter.get()
+    @x_wire_outer.setter
+    def y_wire_outer(self, val: int) -> None:
+        if not isinstance(val, int):
+            print("Value must be an int")
+            return
+        self.controls_information.PVs.ywireouter.put(value = val)
+
+    @property
+    def use_u_wire(self):
+        return self.metadata.PVs.useuwire.get()
+    @use_u_wire.setter
+    def use_u_wire(self, val: int) -> None:
+        if not (val == 1 or val == 0):
+            print("Value must be 0 or 1")
+            return
+        self.controls_information.PVs.useuwire.put(value = val)
+
+    @property
+    def u_wire_inner(self):
+        return self.controls_information.PVs.uwireinner.get()
+    @u_wire_inner.setter
+    def u_wire_inner(self, val: int) -> None:
+        if not isinstance(val, int):
+            print("Value must be an int")
+            return
+        self.controls_information.PVs.uwireinner.put(value = val)
+
+    @property
+    def u_wire_outer(self):
+        return self.controls_information.PVs.uwireouter.get()
+    @u_wire_outer.setter
+    def u_wire_outer(self, val: int) -> None:
+        if not isinstance(val, int):
+            print("Value must be an int")
+            return
+        self.controls_information.PVs.uwireouter.put(value = val)
+
+    @property
+    def initialized(self):
+        return self.controls_information.PVs.enabled.get()
+    @initialized.setter
+    def initalized(self, val) -> None:
+        if not val == 1:
+            print("Value must be 1")
+            return
+        self.controls_information.PVs.initialize.put(value = val)
+
+    @property
+    def homed(self):
+        return self.controls_information.PVs.homed.get()
+
+    @property
+    def position(self):
+        return self.controls_information.PVs.rbv.get()
+    @position.setter
+    def position(self, val: int) -> None:
```

Comment:
I don't know the device very well, but it seems like a lot of these functions would fail if the device is not in its ready state. The position setter in particular seems like it wants to know if the device is ready first. Where that's the case, we should use the `@check_state` decorator, as we do in [the bctrl function in the magnet class](https://github.com/slaclab/lcls-tools/blob/b6949b679df756059f6edd29b83d8dd165d6d4bf/lcls_tools/common/devices/magnet.py#L141).

Reply:
Ask controls group about adding state checks when moving (not scanning) wires.